### PR TITLE
[IE CLDNN] Restore DetectionOutput performance 

### PR DESCRIPTION
**Pull request by yeonbok**
### Details
- Restored DetectionOutput performance 
- Fix some unaligned coding convention 
 

### Tickets:
 - 57310

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 1]
```diff
@@ -33,34 +33,38 @@ namespace {
...
-    explicit detection_output_cpu(const detection_output_node& outer) : outer(outer) {}
+    explicit detection_output_cpu(const detection_output_node& outer) :
+            outer(outer), nms_type(outer.get_primitive()->decrease_label_id ? MXNET : CAFFE) {
+    }
 
-    static void IntersectBBox(const bounding_box& bbox1,
+    static inline void intersect_bbox(const bounding_box& bbox1,
                               const bounding_box& bbox2,
-                              bounding_box& intersectBbox) {
+                              bounding_box& intersect_bbox) {
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:46]: nit: please correct alignment as:
```
    static inline void intersect_bbox(const bounding_box& bbox1,
                                      const bounding_box& bbox2,
                                      bounding_box& intersect_bbox) {
```

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 2]
```diff
@@ -33,34 +33,38 @@ namespace {
 
 /************************ Detection Output CPU ************************/
 struct detection_output_cpu : typed_primitive_impl<detection_output> {
+    enum NMSType {CAFFE, MXNET};
     const detection_output_node& outer;
+    NMSType nms_type;
 
-    explicit detection_output_cpu(const detection_output_node& outer) : outer(outer) {}
+    explicit detection_output_cpu(const detection_output_node& outer) :
+            outer(outer), nms_type(outer.get_primitive()->decrease_label_id ? MXNET : CAFFE) {
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:41]: nit: alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 3]
```diff
@@ -174,15 +178,15 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
         }
     }
 
-    void mxNetNms(const std::vector<std::vector<bounding_box>>& bboxes,
+    void mxnet_nms(const std::vector<std::vector<bounding_box>>& bboxes,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:181]: nit: arguments alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 4]
```diff
@@ -174,15 +178,15 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
...
                   const float nms_threshold,
                   const int top_k,
                   const bool share_location,
                   std::map<int, std::vector<int>>& indices,
                   std::vector<std::pair<float, std::pair<int, int>>>& scoreIndexPairs) {
         std::sort(scoreIndexPairs.begin(),
                     scoreIndexPairs.end(),
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:188]: nit: arguments alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 5]
```diff
@@ -208,23 +212,27 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
         }
     }
 
-    static void caffeNMS(const std::vector<bounding_box>& bboxes,
+    static void caffe_nms(const std::vector<bounding_box>& bboxes,
                          std::vector<std::pair<float, int>>& scores,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:216]: alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 6]
```diff
@@ -237,7 +245,7 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
     }
 
     template <typename T>
-    static bool SortScorePairDescend(const std::pair<float, T>& pair1,
+    static bool comp_score_descend(const std::pair<float, T>& pair1,
                                      const std::pair<float, T>& pair2) {
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:249]: alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 7]
```diff
@@ -305,7 +313,7 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
 
                 std::sort(score_index_pairs.begin(),
                             score_index_pairs.end(),
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:315]: alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 8]
```diff
@@ -499,7 +507,97 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
     }
 
     template <typename dtype>
-    void extract_confidences_per_image(const detection_output_inst& instance,
+    void extract_confidences_per_image_caffe(const detection_output_inst& instance,
+                                       std::vector<std::vector<std::vector<std::pair<float, int>>>>& confidences,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:511]: alignment

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 9]
```diff
@@ -499,7 +507,97 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
...
+                        }
+                        idx += stride;
+                    }
+                }
+            }
+        }
+    }
+
+    template <typename dtype>
+    void extract_confidences_per_image_mxnet(const detection_output_inst& instance,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:600]: alignment

**Review by lznamens — CHANGES_REQUESTED** [review]

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 3 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:181]: fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 9 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:596]: fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 2 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 4 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 6 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 8 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 7 above)
Comment (yeonbok) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: Fixed

**Review by lznamens — APPROVED** [review]
Thanks. Changes look good to me.

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**